src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,50 @@
+package duke.util;
+
+import duke.TaskList;
+import duke.exception.CommandNotFoundException;
+import duke.exception.DukeException;
+import duke.exception.InvalidArgumentException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.ToDo;
+
+public class Parser {
+    public static String parse(String line, TaskList taskList) throws DukeException {
+        String[] input = line.strip().split(" ", 2);
+
+        if (!Constants.TASKS.contains(input[0])) {
+            throw new CommandNotFoundException();
+        }
+
+        if (input[0].equals("bye")) {
+            return "BYE";
+        } else if (input[0].equals("list")) {
+            return taskList.getTasks();
+        } else {
+            if (input.length != 2) {
+                throw new InvalidArgumentException();
+            }
+
+            if (input[0].equals("mark") || input[0].equals("unmark")) {
```

Comment:
Would it be better if we encapsulate `input[0]` into a variable, and use the `switch` statement here instead of a long list of `if` and `else if`?     :)

Reply:
yes, I agree, thanks for the input!